core/src/main/scala/org/allenai/common/cache/QueryCache.scala

```diff
@@ -0,0 +1,78 @@
+package org.allenai.common.cache
+
+import org.allenai.common.Logging
+import redis.clients.jedis.{ Jedis, JedisPool }
+import spray.json._
+
+import scala.collection.mutable
+import scala.concurrent.Future
+
+object QueryCache {
+  val pools: mutable.Map[(String, Int), JedisPool] = new mutable.HashMap[(String, Int), JedisPool]()
```

Comment:
You shouldn't have static mutable state - it isn't thread-safe. You should have your applications manage the pools, since they can do it in a thread-safe manner a lot easier.